ts/components/conversation/AddMentions.tsx

```diff
@@ -0,0 +1,140 @@
+import React from 'react';
+
+import { RenderTextCallbackType } from '../../types/Util';
+import classNames from 'classnames';
+
+declare global {
+  interface Window {
+    lokiPublicChatAPI: any;
+    shortenPubkey: any;
+    pubkeyPattern: any;
+  }
+}
+
+interface MentionProps {
+  key: number;
+  text: string;
+}
+
+interface MentionState {
+  found: any;
+}
+
+class Mention extends React.Component<MentionProps, MentionState> {
+  constructor(props: any) {
+    super(props);
+
+    this.tryRenameMention = this.tryRenameMention.bind(this);
+  }
+
+  public componentWillMount() {
+    const found = this.findMember(this.props.text.slice(1));
+    this.setState({ found });
+
+    this.tryRenameMention();
+    // TODO: give up after some period of time?
+    const intervalHandle = setInterval(this.tryRenameMention, 30000);
+    this.clearOurInterval = this.clearOurInterval.bind(this, intervalHandle);
+  }
+
+  public componentWillUnmount() {
+    this.clearOurInterval(null);
+  }
+
+  public render() {
+    if (this.state.found) {
+      // TODO: We don't have to search the database of message just to know that the message is for us!
+      const us =
+        this.state.found.authorPhoneNumber === window.lokiPublicChatAPI.ourKey;
+      const className = classNames(
+        'mention-profile-name',
+        us && 'mention-profile-name-us'
+      );
+
+      return (
+        <span className={className}>{this.state.found.authorProfileName}</span>
+      );
+    } else {
+      return (
+        <span className="mention-profile-name">
+          {window.shortenPubkey(this.props.text)}
+        </span>
+      );
+    }
+  }
+
+  private clearOurInterval(handle: any) {
+    clearInterval(handle);
```

Comment:
can you do `clearInterval(this.intervalHandle);` and not accept any parameter?
`clearOurInterval` is only called with `null`

Reply:
Even though it is called with `null` it works (because of bind)! I don't think you can add arbitrary data to `this` in tsx, that's why I had to find another way...

Reply:
Ah right, I see now. I didn't seen that you had bound `intervalHandle` :)
`this.clearOurInterval = this.clearOurInterval.bind(this, intervalHandle);`

I think you can simply declare a member variable to hold the interval handle?

Reply:
Updated as per discussion on discord!